src/edit.js

```diff
@@ -9,6 +9,57 @@ import 'codemirror/addon/search/searchcursor';
 import 'codemirror/addon/search/jump-to-line';
 import 'codemirror/addon/dialog/dialog';
 import 'codemirror/addon/dialog/dialog.css';
+import 'codemirror/addon/mode/overlay';
+import 'codemirror/addon/mode/simple';
+
+CodeMirror.defineSimpleMode('wcms', {
+    // detect a Wcms markup then pass to 'wcms' mode
+    start: [
+        {
+            regex: /%(?=(HEADER|NAV|ASIDE|MAIN|FOOTER|SUMMARY|LIST|MEDIA)(\?[^\s]*)?%)/,
+            token: 'wcms',
+            next: 'wcms',
+        },
+    ],
+    // 'wcms' mode, for each macro, if there is parameters, pass to its associated mode
+    wcms: [
+        {
+            regex: /(HEADER|NAV|ASIDE|MAIN|FOOTER)\?/,
+            token: 'wcms',
+            next: 'element',
+        },
+        { regex: /SUMMARY\?/, token: 'wcms', next: 'summary' },
+        { regex: /LIST\?/, token: 'wcms', next: 'list' },
+        { regex: /MEDIA\?/, token: 'wcms', next: 'media' },
+        { regex: /[^&]*&/, token: 'wcms', pop: true },
+        { regex: /.*%/, token: 'wcms', next: 'start' },
+    ],
+    // 'element' mode, parameters' keywords of 'element' macros
+    element: [{ regex: null, push: 'wcms' }],
+    // 'summary' mode, parameters' keywords of the 'summary' macro
+    summary: [
+        { regex: /min|max/, token: 'keyword', push: 'wcms' },
```

Comment:
> Do you think that it could be not too complicated to add a special class for keywords that are inside a w specific syntax

This is extremely easy, you just have to change the mode of your keywords tokens. Here I used the already existing `keyword` mode, so tthis token will have the class `cm-keyword`. You can see that I used `wcms` at some places. This is why these tokens have the `cm-wcms` class.

Reply:
OOoohhhhhh Thank You